library/psa_util.c

```diff
@@ -262,12 +261,8 @@ mbedtls_ecp_group_id mbedtls_ecc_group_from_psa(psa_ecc_family_t curve,
 #endif
 #if defined(PSA_WANT_ECC_SECP_R1_521)
                 case 521:
-                    return MBEDTLS_ECP_DP_SECP521R1;
                 case 528:
-                    if (bits_is_sloppy) {
-                        return MBEDTLS_ECP_DP_SECP521R1;
-                    }
-                    break;
+                    return MBEDTLS_ECP_DP_SECP521R1;
```

Comment:
There was some ambiguity in the task description: removing `bits_is_sloppy` can be done either by behaving as if it's always true or always false. You chose the former. Can you explain why?

Also, this is something we want to document if we keep it that way. (Which is actually an argument against maintaining "sloppy" behaviour: public functions should be simple to document.)

Edit (after reading more of the code): actually, since there is only one place that potentially used `bits_is_sloppy == 1`, it would seem more natural to me for the function to always behave as false, and then for the only call site that needed sloppy behaviour to have a special case here. (Could be an explicit check for one of the two (family,size) pairs concerned, or perhaps we would include that in `check_ecc_parameters()`: for example instead of taking `int allow_bit_size_roundup` it could take `size_t *actual_bits` - if `NULL`, then checking is strict, if non-`NULL` checking is more relaxed and `*actual_bits` is updated upon return.

Wdyt?